api/src/users/services/users.service.ts

```diff
@@ -1,85 +1,140 @@
     }
 
     async findOne(id: number) {
-        return this.userRepository
-            .findOneOrFail({
-                where: { id },
-                relations: { userAvatar: true }
-            })
-            .catch(() => {
+        return this.userRepository.findOneByOrFail({ id }).catch(error => {
+            if (error instanceof EntityNotFoundError)
                 throw new NotFoundException();
+
+            throw error;
+        });
+    }
+
+    async findOneByIdAndUserJwtId(id: number, userId: number) {
+        const user = await this.userRepository
+            .findOneByOrFail({ id })
+            .catch(error => {
+                if (error instanceof EntityNotFoundError)
+                    throw new NotFoundException();
+
+                throw error;
             });
+
+        if (!this.isDevelopment && user.id !== userId)
+            throw new ForbiddenException("You are not the owner of account");
+
+        return user;
     }
 
     async findOneByUsername(username: string) {
-        return this.userRepository.findOneByOrFail({ username }).catch(() => {
-            throw new NotFoundException();
-        });
+        return this.userRepository
+            .findOneByOrFail({ username })
+            .catch(error => {
+                if (error instanceof EntityNotFoundError)
+                    throw new NotFoundException();
+
+                throw error;
+            });
     }
 
     async selectPassword(username: string) {
         return this.userRepository
             .findOneOrFail({
                 where: { username },
-                select: ["password"]
+                select: ["id", "password"]
             })
-            .catch(() => {
-                throw new NotFoundException();
+            .catch(error => {
+                if (error instanceof EntityNotFoundError)
+                    throw new NotFoundException("Username don't exist");
+
+                throw error;
             });
     }
 
     async delete(id: number) {
-        const result = (await this.userRepository.delete(id)).affected;
-
-        if (!result) throw new NotFoundException();
+        return this.userRepository.delete(id);
     }
 
-    async update(id: number, updateUserDto: UpdateUserDto) {
+    async update(
+        id: number,
+        userId: number,
+        updateUserDto: UpdateUserDto,
+        file: IUploadedFile
+    ) {
+        const user = await this.findOneByIdAndUserJwtId(id, userId);
+
         updateUserDto.username = updateUserDto.username
             .replace(/\s/g, "")
             .toLowerCase();
 
-        const user = await this.userRepository.findOneBy({
-            username: updateUserDto.username
-        });
+        const avatar = user.userAvatar;
 
-        if (user && user.id !== id)
-            throw new ConflictException("Duplicated username");
+        if (file)
+            updateUserDto.userAvatar =
+                await this.usersAvatarService.addUserAvatarFile(
```

Comment:
I already told You about this function but You probably forgot. `addUserAvatarFile` is wrong because it has side effect. Look at this function as a API. You see `addUserAvatarFile` so you expect that this function only add and when you see implementation. There is also deleting stuff. 

For me it should be somthing like
```
    async addUserAvatarFile(
        id: number,
        file: IUploadedFile,
    ) {
        const name = file.originalname;
        const extension = file.originalname.split(".").pop();
        const sourcePath = file.path;
        
        return this.usersAvatarRepository.save({ name, sourcePath, extension });
    }
```
And 
```
    async update(
        id: number,
        userId: number,
        updateUserDto: UpdateUserDto,
        file: IUploadedFile
    ) {
        const user = await this.findOneByIdAndUserJwtId(id, userId);

        if (user.userAvatar){
            await this.usersAvatarService.delete(
                user.userAvatar.id,
                user.userAvatar.sourcePath
            );
        }
        
        if (file) {
            updateUserDto.userAvatar =
                await this.usersAvatarService.addUserAvatarFile(
                    id,
                    file,
                );
        } else {
            updateUserDto.userAvatar = null;
        }

        await this.userRepository.update(id, updateUserDto).catch(error => {
            if (error instanceof QueryFailedError)
                throw new ConflictException("Duplicated username");

            throw error;
        });
    }
```
You could also refactor `async delete(awatar: AvatarEntityOrDto)` and all calls

Reply:
ye, that's cause i don't want to check if the file exist on server, cause it's take that much resources as remove and and, because i need to compare that.

But ye, you're right, side effect it's unnecessary i must refactor it.

Reply:
done